datafusion/core/src/lib.rs

```diff
@@ -744,8 +744,16 @@ pub mod physical_planner;
 pub mod prelude;
 pub mod scalar;
 
-// re-export dependencies from arrow-rs to minimize version maintenance for crate users
+// re-export dependencies to minimize version maintenance for crate users
 pub use arrow;
+/// Re-export of the [`object_store`] crate so users can access object store
+/// types without depending on `object_store` directly.
```

Comment:
I think rationale for exporting arrow, parquet is similar to the one for object store.
We can make all similar  & short

```
// Re-export dependencies that are part of DataFusion public API (e.g. via DataFusionError)
```

No need for code example. It doesn't help understand the _why_.